### ISSUE TITLE
Cant't deserialize global

### ISSUE THREAD

**Issue by LordMZTE** [khvzak/mlua#74]
## Issue
It doesn't seem to be possible, for example this code
```rs
use mlua::{Lua, LuaSerdeExt, Value};
use serde::Deserialize;

fn main() {
    let lua = Lua::new();

    lua.load(
        r#"
        hello = "Hello, world!"
        "#,
    )
    .exec()
    .expect("exec failed");

    let global = lua
        .from_value::<Global>(
            // This is necessary to give from_value the expected type
            Value::Table(lua.globals()),
        )
        .expect("Deserialize fail");

    dbg!(global);
}

#[derive(Debug, Deserialize)]
struct Global {
    hello: String,
}
```
fails with
```
thread 'main' panicked at 'Deserialize fail: DeserializeError("invalid value type")', src/main.rs:20:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Is this a bug, or is there some reason that global can't be deserialized, and if so, what would i do about it?

## Versions
- mlua: 0.6.2
- lua runtime: can reproduce with lua54 and luajit

**Comment by khvzak** [comment]
Thanks for the report.
This is probably a bug. A Lua global table contains many other object (like `require` function), and the deserializer returns error because don't know how to deserialize such elements.
I think I need to introduce `DeserializeOptions` struct to configure deserializer behaviour (and be relaxed by default).

**Comment by khvzak** [comment]
It's fixed in v0.6.3 and requires to use new function `Lua::to_value_with` with the corresponding options:
```
    let options = mlua::DeserializeOptions::new()
        .deny_recursive_tables(false)
        .deny_unsupported_types(false);
    let global = lua
        .from_value_with::<Global>(
            // This is necessary to give from_value the expected type
            Value::Table(lua.globals()),
            options,
        )
        .expect("Deserialize fail");
```

**Comment by LordMZTE** [comment]
Thanks! This fixes my issue!
